packages/amplify_datastore/ios/Classes/SwiftAmplifyDataStorePlugin.swift

```diff
@@ -81,10 +81,19 @@ public class SwiftAmplifyDataStorePlugin: NSObject, FlutterPlugin {
     }
     
     private func onConfigureDataStore(args: [String: Any], result: @escaping FlutterResult) {
-        
-        guard let modelSchemaList = args["modelSchemas"] as? [[String: Any]] else {
-            result(false)
-            return //TODO
+
+        guard let modelSchemaList = args["modelSchemas"] as? [[String: Any]],
+              let modelProviderVersion = args["modelProviderVersion"] as? String else {
+
+            FlutterDataStoreErrorHandler.handleDataStoreError(
+                error:
+                    DataStoreError.decodingError(
+                        "Received invalid request from Dart, modelSchemas and/or modelProviderVersion are not available. Request: " + args.description,
+                        "Check the values that are being passed from Dart."
```

Comment:
Nit: Since this is a Flutter handler I wonder if we should just reference Flutter directly rather than Dart